flexmeasures/data/models/planning/storage.py

```diff
@@ -235,6 +236,26 @@ def _prepare(self, skip_validation: bool = False) -> tuple:  # noqa: C901
                 method="upper",
             )
 
+        usage_forecast = self.flex_model.get("usage_forecast", [])
+
+        all_usage_forecast = []
+
+        for component in usage_forecast:
+            all_usage_forecast.append(
+                get_continous_series_sensor_or_quantity(
+                    quantity_or_sensor=component,
+                    actuator=sensor,
+                    target_unit="MWh",
+                    query_window=(start, end),
+                    resolution=resolution,
+                    beliefs_before=belief_time,
+                )
+            )
+        if len(all_usage_forecast) > 0:
+            all_usage_forecast = pd.concat(all_usage_forecast, axis=1)
+
+            device_constraints[0]["usage forecast"] = all_usage_forecast.sum(1)
```

Comment:
The usage forecast may still need to be scaled using the sensor resolution divided over an hour. See how other SoC constraints are added.

Reply:
I believe this should be added:

```
 * timedelta(hours=1) / resolution
 ```

The SoC units within the device_scheduler are not `MWh`, but rather `MW * <event_resolution>`. (This is also the reason that SOC constraint values in the pyomo logs look different than those defined in the problem statement. The SoC constraints are scaled in lines 758-759 of `add_storage_constraints` by the way.)

And then to make the `test_battery_usage_forecast_sensor` pass again, the usage forecast should be 2 MWh for each 15 minutes long event. Alternatively, change the usage forecast sensor to MW and set its values to 8 MW for each 15 minutes long event.

Reply:
Thanks for the guidance. Indeed, we need to take the event resolution into consideration . I've introduced some changes to convert from one resolution to the target (the one of the `power` sensor)  and an extra case to check that it works if we define the same energy in a different way. 

In the test I'm defining a constant usage forecast of 2 MWh every 15min as:
- Energy sensor with resolution 15min and constant value 2 MWh
- Energy sensor with resolution 1h and constant value 8 MWh